### PR TITLE
feat(bom): Script to 'promote' BOM for release.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

File: dev/promote_bom.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  def __init__(self, options):
+    self.__rc_version = options.rc_version
+    self.__bom_dict = {}
+    super(BomPromoter, self).__init__(options)
+
+  def __unpack_bom(self):
+    """Load the release candidate BOM into memory.
+    """
+    # TODO(jacobkiefer): Replace this command with hal.
+    bom_yaml_string = run_quick('cat 0.3.0-184.yml', echo=False).stdout.strip()
```
Comment (lwander) [inline dev/promote_bom.py]: You can replace this now if you like, `hal versions bom 0.3.0-184 --color false`

Comment (lwander) [inline dev/promote_bom.py]: Released in halyard 0.10.0

**Comment by jtk54** [comment]
Updated to pull the release candidate BOM using `hal`.

File: dev/promote_bom.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    cls.init_argument_parser(parser)
+    options = parser.parse_args()
+
+    bom_promoter = cls(options)
+    bom_promoter.promote_bom()
+
+  @classmethod
+  def init_argument_parser(cls, parser):
+    """Initialize command-line arguments."""
+    parser.add_argument(
```
Comment (ewiseblatt) [inline dev/promote_bom.py]: It seems this is required=True

Comment (jtk54) [inline dev/promote_bom.py]: Yep, added that in.

**Pull request merged**